src/swagger.json

```diff
@@ -0,0 +1,236 @@
+{
+  "openapi": "3.0.1",
+  "info": {
+    "version": "1.0.0",
+    "title": "Community Hub API",
+    "description": "Bridge Project"
+  },
+
+  "servers": [
+    {
+      "url": "http://localhost:3000",
+      "description": "local server"
+    }
+  ],
+  "tags": [
+    {
+      "name": "CRUD Operations"
+    }
+  ],
+
+  "paths": {
+    "/events": {
+      "get": {
+        "tags": ["CRUD Operations"],
+        "description": "Get all events.",
+        "operationId": "getEvents",
+        "responses": {
+          "200": {
+            "description": "Events were obtained",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Event"
+                }
+              }
+            }
+          }
+        }
+      },
+      "post": {
+        "tags": ["CRUD Operations"],
+        "description": "Post a new event.",
+        "operationId": "postEvent",
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/Event"
+              }
+            }
+          }
+        },
+        "responses": {
+          "201": {
+            "description": "New event was created.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Event"
+                }
+              }
+            }
+          },
+          "422": {
+            "description": "Invalid paramaters",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Event"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+
+    "/events:id": {
+      "put": {
+        "tags": ["CRUD Operations"],
+        "description": "Update an event",
+        "operationId": "updateEvent",
+        "paramaters": [
+          {
+            "name": "id",
+            "in": "path",
+            "schema": {
+              "properties": {
+                "id": {
+                  "type": "string"
+                }
+              }
+            },
+            "required": "true"
+          },
+          {
+            "name": "event",
```

Comment:
This didn't seem to work at all, and I am cursing myself for not using YAML but I was already too deep with JSON! Looking for an easy to understand guide/explanation on how to indicate optional params in the request body.

Reply:
Actually, you don't need to indicate that a parameter is optional with openapi. All parameters are optional be default, so I think you've got it right here. 

More info: https://swagger.io/docs/specification/describing-parameters/

Reply:
If you really would prefer to use YAML, It's not too late! Json can be converted into YAML very easily, since the latter is a 'superset'.  So you can probably find a JSON -> YAML converter tool easily, and then use the format you prefer.